spec/models/tag_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+describe Tag do
+
+  it "decreases tag_count when a debate is hidden" do
+    debate = create(:debate)
+    tag = create(:tag)
+    tagging = create(:tagging, tag: tag, taggable: debate)
+
+    expect(tag.taggings_count).to eq(1)
+
+    debate.update(hidden_at: Time.now)
+
+    tag.reload
+    expect(tag.taggings_count).to eq(0)
+  end
+
+  it "decreases tag_count when a proposal is hidden" do
```

Comment:
This is exactly the same test as above but for proposal instead debate, what about using [shared examples](https://www.relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples)? :thinking: